src/Adapter/GeneralConfiguration.php

```diff
@@ -38,9 +39,15 @@ class GeneralConfiguration implements DataConfigurationInterface
      */
     private $configuration;
 
-    public function __construct(Configuration $configuration)
+    /**
+     * @var Cookie
+     */
+    private $cookie;
+
+    public function __construct(Configuration $configuration, Cookie $cookie)
```

Comment:
```suggestion
    public function __construct(Configuration $configuration, Cookie $cookie = null)
```

Else it's a BC break ?

Reply:
Yes it's a BC, forget to edit the PR description. We can't have $cookie = null, otherwise the SameSite configuration will not work as expected.